core/src/test/java/io/cucumber/core/plugin/PluginFactoryTest.java

```diff
@@ -49,12 +49,16 @@
     void instantiates_junit_plugin_with_file_arg() {
         Object plugin = fc.create(parse("junit:" + tmp.resolve("cucumber.xml")));
         assertThat(plugin.getClass(), is(equalTo(JUnitFormatter.class)));
+
+        release_resources(plugin);
     }
 
     @Test
     void instantiates_rerun_plugin_with_file_arg() {
         Object plugin = fc.create(parse("rerun:" + tmp.resolve("rerun.txt")));
         assertThat(plugin.getClass(), is(equalTo(RerunFormatter.class)));
+
+        release_resources(plugin);
```

Comment:
This is quite repetitive. Consider wrapping `fc.create` and releasing the resources before returning the created plugin.

Reply:
fc is the instance under test and personally I prefer to see the tested call in the test directly. But I moved it into an afterEach hook, looks less cluttered